tests/tasks/test_bfdtask.py

```diff
@@ -0,0 +1,97 @@
+import pytest
+
+from zino.config.models import PollDevice
+from zino.oid import OID
+from zino.state import ZinoState
+from zino.statemodels import BFDEvent, BFDSessState, BFDState, Port
+from zino.tasks.bfdtask import BFDTask
+
+
+@pytest.fixture()
+def device_port():
+    """Port related to the BFD state at the simulated device"""
+    yield Port(ifindex=1, ifdescr="xe-1/2/0.0")
+
+
+@pytest.fixture()
+def bfd_state():
+    """Represents the BFD state at the simulated device"""
+    yield BFDState(session_state=BFDSessState.UP, session_index=4524)
+
+
+@pytest.fixture()
+def bfd_task(snmpsim, snmp_test_port, device_port):
+    device = PollDevice(
+        name="buick.lab.example.org",
+        address="127.0.0.1",
+        port=snmp_test_port,
+        community="juniper-bfd-up",
+    )
+    state = ZinoState()
+    device_state = state.devices.get(device_name=device.name)
+    device_state.enterprise_id = 2636
+
+    # this effectively makes the device_port fixture a shortcut for accessing
+    # the port that will change bfd_state when the task is run
+    device_state.ports[device_port.ifindex] = device_port
+    task = BFDTask(device, state)
+    yield task
+
+
```

Comment:
Advocating for the step-down rule, I would want the test code to come first, and the fixtures they depend on to come later on.

An old ref for new devs:
![the-stepdown-rule](https://github.com/Uninett/zino/assets/100995/0a3e7573-0cf9-4965-b072-295392e6af5a)

While it is impossible to adhere 100% to the stepdown rule in a non-compiled language like Python (definitions that are needed at import-time must be defined first), I try to stick to this rule as much as possible, so that I can open any source code file and read the highest-level concepts first.